subprojects/gradle/impact-shared/src/main/java/com/avito/impact/ModifiedProjectsFinder.kt

```diff
@@ -27,46 +47,10 @@ class ModifiedProjectsFinder(
      *
      * @return only modified projects by [configurationType]
      */
-    fun findModifiedProjects(configurationType: ConfigurationType? = null): Set<ModifiedProject> {
-        val reportTypes = if (configurationType == null) {
-            ConfigurationType.values()
-        } else {
-            setOf(configurationType)
+    override fun modifiedProjects(configurationType: ConfigurationType): Set<ModifiedProject> {
+        return findProjects(rootProject) {
+            it.internalModule.isModified(configurationType)
         }
-        return reportTypes
-            .flatMap { type ->
-                findProjects(rootProject, type) {
-                    it.internalModule.isModified(type)
-                }
-            }.toSet()
-    }
-
-    @Deprecated(
-        "Используется только для поиска по ReportType.ANDROID_TESTS. " +
-            "Оптимизация для UI тестов, явно игнорируем изменения в реализации, чтобы не сваливаться всегда в fallback"
-    )
-    fun findModifiedProjectsWithoutDependencyToAnotherConfigurations(
-        configurationType: ConfigurationType
-    ): Set<ModifiedProject> =
-        findProjects(rootProject, configurationType) {
-            it.internalModule.getConfiguration(configurationType).let { configuration ->
-                configuration.dependencies.any { dependency -> dependency.isModified }
-                    || configuration.hasChangedFiles
-            }
-        }
-
-    private fun findProjects(
-        rootProject: Project,
-        configurationType: ConfigurationType,
-        predicate: (project: Project) -> Boolean
-    ): Set<ModifiedProject> {
```

Comment:
Here was a mistake. The function could return all projects as modified.

```
20 line: findProjects(rootProject, type, predicate = { true })
```

This is why I've introduced an explicit API to get all analyzed projects `allProjects()`.